exporter/azuremonitorexporter/connection_string_parser.go

```diff
@@ -75,6 +75,6 @@ func getIngestionURL(ingestionEndpoint string) string {
 		ingestionURL, _ = url.Parse(DefaultIngestionEndpoint)
 	}
 
-	ingestionURL.Path = path.Join(ingestionURL.Path, "/v2/track")
+	ingestionURL.Path = path.Join(ingestionURL.Path, "/v2.1/track")
```

Comment:
What's the relevance of this change to the documentation update? If this is a functionality change it may be good to have a separate PR.

Reply:
There is no separate document specifically addressing Application Insights versions. However, the Azure Monitor service schema, which contains the details, can be found here: [Service Schema](https://github.com/Azure/azure-rest-api-specs/blob/6013319f5a9f2fc613f762510e9dd46403b922bf/specification/applicationinsights/data-plane/Monitor.Exporters/preview/v2.1/swagger.json#L9)

This topic is also discussed in the troubleshooting document: [Troubleshooting Document](https://learn.microsoft.com/en-us/azure/azure-monitor/app/azure-ad-authentication?tabs=net#http11-400-authentication-not-supported)

This is a follow-up to work completed in PR #28854. The Instrumentation Key will soon be deprecated, along with specifying the endpoint. This PR includes a documentation update and transfers the responsibility of specifying the service version to the code itself. This approach has been adopted for Azure Monitor exporters in .NET, Java, Node.js, and Python.

I have included both the documentation and this change in the same PR, as they are closely related. For Azure Active Directory (AAD) authentication, it is mandatory to use v2.1, as v2.0 will not be supported. Previous documentation required customers to hard-code this section.

Reference: On March 31, 2025, support for instrumentation key ingestion will end. Instrumentation key ingestion will continue to work, but we'll no longer provide updates or support for the feature. [Transition to connection strings](https://learn.microsoft.com/en-in/azure/azure-monitor/app/migrate-from-instrumentation-keys-to-connection-strings) to take advantage of [new capabilities](https://learn.microsoft.com/en-in/azure/azure-monitor/app/migrate-from-instrumentation-keys-to-connection-strings#new-capabilities).

Reply:
this is changing the behaviour though, should it not be documented in a changelog entry for end users?

Reply:
Added changelog entry.